client/src/components/MainPage/PreviewCategories/PreviewCategories.jsx

```diff
@@ -0,0 +1,341 @@
+import {
+  Block,
+  BlockPicture,
+  BlockPictureDescription,
+  PictureDescription,
+  Button,
+  BlockPictureImg,
+} from "./PreviewCategories.styled";
+import { useState, useEffect } from "react";
+
+const PreviewCategories = ({
+  pictureDescription,
+  button,
+  breakfast,
+  miscellaneous,
+  chicken,
+  desserts,
+}) => {
+  const [breakfastData, setBreakfastData] = useState([]);
+  const [miscellaneousData, setMiscellaneousData] = useState([]);
+  const [chickenData, setChickenData] = useState([]);
+  const [dessertsData, setDessertsData] = useState([]);
+
+  // do poprawy
+
+  useEffect(() => {
+    fetch("http://localhost:5000/recipes/main-page")
+      .then((response) => response.json())
+      .then((data) => {
+        setBreakfastData(data.Breakfast);
+        setMiscellaneousData(data.Miscellaneous);
+        setChickenData(data.Chicken);
+        setDessertsData(data.Dessert);
+      })
+      .catch((error) => console.error("Błąd pobierania danych:", error));
+  }, []);
+
+  const firstBreakfast = breakfastData.length > 0 ? breakfastData[0] : null;
+  const twoBreakfast = breakfastData.length > 0 ? breakfastData[1] : null;
+  const threeBreakfast = breakfastData.length > 0 ? breakfastData[2] : null;
+  const fourBreakfast = breakfastData.length > 0 ? breakfastData[3] : null;
+
+  const firstMiscellaneous =
+    miscellaneousData.length > 0 ? miscellaneousData[0] : null;
+  const twoMiscellaneous =
+    miscellaneousData.length > 0 ? miscellaneousData[1] : null;
+  const threeMiscellaneous =
+    miscellaneousData.length > 0 ? miscellaneousData[2] : null;
+  const fourMiscellaneous =
+    miscellaneousData.length > 0 ? miscellaneousData[3] : null;
+
+  const firstChicken = chickenData.length > 0 ? chickenData[0] : null;
+  const twoChicken = chickenData.length > 0 ? chickenData[1] : null;
+  const threeChicken = chickenData.length > 0 ? chickenData[2] : null;
+  const fourChicken = chickenData.length > 0 ? chickenData[3] : null;
+
+  const firstDesserts = dessertsData.length > 0 ? dessertsData[0] : null;
+  const twoDesserts = dessertsData.length > 0 ? dessertsData[1] : null;
+  const threeDesserts = dessertsData.length > 0 ? dessertsData[2] : null;
+  const fourDesserts = dessertsData.length > 0 ? dessertsData[3] : null;
```

Comment:
ta sekcja do poprawy, jeżeli faktycznie potrzebujesz tyle zmiennych to zrób do tego funkcję, bo tego jest za dużo